### ISSUE TITLE
rollup openvpn 2.4 in usg3

### ISSUE THREAD

**Issue by EpicTMC** [unifi-utilities/unifios-utilities#477]
Version:      v4.4.57
Build ID:     5578372
Build on:     01/12/23 08:23
Copyright:    2012-2020 Ubiquiti, Inc.
HW model:     UniFi-Gateway-3

This version still hosts an older and outdated OpenVPN client version (2.3.2) which is unsupported for various VPN providers and shows these errors in the log when trying to connect:

USG openvpn[3804]: TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
USG openvpn[3804]: TLS Error: TLS handshake failed
USG openvpn[3804]: SIGUSR1[soft,tls-error] received, process restarting

We need to rollup to OpenVPN version 2.4.x in order to have vpn access through the usg3

**Comment by sprocktech** [comment]
WireGuard works on the USG. Use that instead of OpenVPN if you have the option.

https://github.com/WireGuard/wireguard-vyatta-ubnt/wiki/EdgeOS-and-Unifi-Gateway

**Comment by EpicTMC** [comment]
> WireGuard works on the USG. Use that instead of OpenVPN if you have the option.
> 
> https://github.com/WireGuard/wireguard-vyatta-ubnt/wiki/EdgeOS-and-Unifi-Gateway

thanks, i have seen this option but as i already have several ssids that point over different nordvpn configs, and the json already built to openvpn i havent dived to much into wireguard (yet)

**Comment by sprocktech** [comment]
Also, this is not an official repo from the vendor. It's community supported and made for "UniFi OS" hardware, which doesn't include the USG. I would suggest either looking at ways to upgrade OpenVPN on your USG yourself, or switch to WG.

**Comment by EpicTMC** [comment]
> upgrade OpenVPN on your USG yourself, or switch to WG.

neither simple haha

**Comment by EpicTMC** [comment]
> https://community.ui.com/questions/Connect-your-Unifi-USG-to-a-VPN-Provider-Includes-config-for-optional-extra-wireless-networks-per-c/67d49971-1841-4ccc-b0a1-71e3aaf6eb6a#answer/e136c826-fdc7-41dc-80fa-80229a8e0c3a

thanks ill have a look. 
do you have anything similar to this link saved for getting wiregurad running with nord (or any other vpn provider)?

**Comment by sprocktech** [comment]
No, but you should be able to adjust what you have, adding the appropriate WG interfaces, and then replacing your OpenVPN interfaces like `vtun0` with WG interfaces like `wg0` in your firewall, NAT, and static routing config.
